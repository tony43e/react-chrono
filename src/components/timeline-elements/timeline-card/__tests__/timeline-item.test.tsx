import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Item from '../timeline-card';

const onClick = jest.fn();
const autoScroll = jest.fn();

const id = 'test_azsxcs1212_test';

const Horizontal = (
  <Item
    active
    contentText="This is a test content"
    id="34edc"
    mode="HORIZONTAL"
    onClick={() => onClick('CC22CC')}
    autoScroll={autoScroll}
    position="top"
    title="test title"
    wrapperId={id}
  />
);

const Vertical = (
  <Item
    active
    contentText="This is a test content"
    id="34ed9c"
    mode="VERTICAL"
    onClick={onClick}
    autoScroll={autoScroll}
    position="bottom"
    title="test title"
    wrapperId={id}
  />
);

test('Test Timeline Item render - Horizontal', () => {
  render(Horizontal);

  const element = screen.getByTestId('timeline-item');

  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

test('Test Timeline Item render - Vertical', () => {
  render(Vertical);

  const element = screen.getByTestId('timeline-item');

  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

test('Test Timeline Item render - Horizontal Class', () => {
  render(Horizontal);

  const element = screen.getByTestId('timeline-item');
  const titleElement = screen.getByTestId('timeline-title');

  expect(element).toHaveClass('horizontal');
  expect(titleElement).toHaveClass('horizontal top');
});

test('Test Timeline Item render - Vertical Class', () => {
  render(Vertical);
  const element = screen.getByTestId('timeline-item');
  const titleElement = screen.getByTestId('timeline-title');

  expect(element).toHaveClass('vertical');
  expect(titleElement).toHaveClass('vertical bottom');
});

test('Test Timeline Item - Handler', () => {
  render(Horizontal);
  const element = screen.getByTestId('timeline-circle');

  fireEvent.click(element);
  expect(onClick).toBeCalled();
  expect(onClick).toBeCalledWith('CC22CC');
});
