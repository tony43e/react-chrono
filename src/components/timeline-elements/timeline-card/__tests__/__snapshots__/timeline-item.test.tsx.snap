// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Timeline Item render - Horizontal 1`] = `
<div
  class="horizontal css-16eji5t"
  data-testid="timeline-item"
>
  <div
    class="css-1p18t0t"
  >
    <div
      class="horizontal active css-1a01u17"
      data-testid="timeline-circle"
    />
  </div>
  <div
    class="horizontal top css-lzd26m"
    data-testid="timeline-title"
  >
    <div
      class="active css-1w5m28t"
      title="test title"
    >
      test title
    </div>
  </div>
</div>
`;

exports[`Test Timeline Item render - Vertical 1`] = `
<div
  class="vertical css-16eji5t"
  data-testid="timeline-item"
>
  <div
    class="css-1p18t0t"
  >
    <div
      class="vertical active css-1a01u17"
      data-testid="timeline-circle"
    />
  </div>
  <div
    class="vertical css-nyflco"
  >
    <div
      class="vertical bottom css-lzd26m"
      data-testid="timeline-title"
    >
      <div
        class="active css-1w5m28t"
        title="test title"
      >
        test title
      </div>
    </div>
    <div
      class="active css-1a8fhdt"
    >
      <span
        class="css-4hae14"
      >
        This is a test content
      </span>
      <div
        class="show-less css-hrwi5d"
      />
      <span
        class="show-more css-1rml49d"
        role="button"
      >
        show more
      </span>
    </div>
  </div>
</div>
`;
